opm/simulators/linalg/WellOperators.hpp

```diff
@@ -0,0 +1,239 @@
+/*
+  Copyright 2016 IRIS AS
+  Copyright 2019, 2020 Equinor ASA
+  Copyright 2020 SINTEF
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WELLOPERATORS_HEADER_INCLUDED
+#define OPM_WELLOPERATORS_HEADER_INCLUDED
+
+#include <dune/istl/operators.hh>
+
+
+namespace Opm
+{
+
+
+//=====================================================================
+// Implementation for ISTL-matrix based operators
+// Note: the classes WellModelMatrixAdapter and
+// WellModelGhostLastMatrixAdapter were moved from ISTLSolverEbos.hpp
+// and subsequently modified.
+//=====================================================================
+
+// This class is intended to hide the actual type of the well model
+// (which depends on a TypeTag) in the WellModelMatrixAdapter, so that
+// we do not need to instantiate that class (and classes using it) for
```

Comment:
I don't understand this sentence. If the WellModel depends on the TypeTag, then it is still intantiated for it. We just cannot access it here. IMHO this is a good thing a good thing. Is that the purpose? Otherwise WellModel could be a LinearOperator itself, too.

Reply:
Got it. Sounds good.